docs/commerce/payment/extend_payment.md

```diff
@@ -0,0 +1,120 @@
+---
+description: Extend Payment with custom payment method type.
+edition: commerce
+---
+
+# Extend Payment
+
+There are different ways you can extend your Payment module implementation. 
+One of them is to create a custom payment method type. 
+The other is attaching custom data to a payment.
+
+You can also [customize the payment processing workflow](configure_payment.md#custom_payment_workflows).
+
+## Create custom payment method type
+
+If your application needs payment methods of other type than the default `offline` one, you must create custom payment method types. 
```

Comment:
Maybe add a sentence t the tune of: the example below shows this on the example of PayPal...